### ISSUE TITLE
the weights of Threshold Encoder be NaN when training

### ISSUE THREAD

**Issue by s1702319** [taohan10200/IIM#6]
Hello, your work is interesting and inspired. I'm trying to re-implement your paper using keras, but the weights of threshold encoder would be nan after some epochs. Have this phenomenon happened to you, or could you give some suggestions?

thank you~

**Comment by taohan10200** [comment]
This problem occurs in some datasets where you can lower the learning rate of the threshold encoder, such as 1e-7.

Thanks for your attention!

**Comment by s1702319** [comment]
> This problem occurs in some datasets where you can lower the learning rate of the threshold encoder, such as 1e-7.
> 
> Thanks for your attention!

Thanks for your reply, I will try it.
